app/src/pages/HomePageContent.vue

```diff
@@ -0,0 +1,267 @@
+<script setup lang="ts">
+import HorizontalContentTileCollection from "@/components/content/HorizontalContentTileCollection.vue";
+import { computed, ref, watch, type Ref } from "vue";
+import { type ContentDto, DocType, TagType, db } from "luminary-shared";
+import { useAuth0 } from "@auth0/auth0-vue";
+import { appLanguageIdAsRef } from "@/globalConfig";
+import IgnorePagePadding from "@/components/IgnorePagePadding.vue";
+
+type ContentByCategory = {
+    category: ContentDto;
+    latestContentDate: number;
+    content: Array<ContentDto>;
+};
+
+/**
+ * Sort content by category
+ * @param content
+ * @param categories
+ * @returns a Vue ref of ContentByCategory[]
+ */
+const contentByCategory = (
+    content: Ref<ContentDto[]>,
+    categories: Ref<ContentDto[]>,
+): Ref<ContentByCategory[]> => {
+    const result = ref<ContentByCategory[]>([]);
+
+    watch([content, categories], ([_content, _categories]) => {
```

Comment:
This could also be written with a `watchEffect`, then the dependencies don't need to be watched manually and you forego the need for the extra underscored variables.